File: app/Models/Ejercicio.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Ejercicio extends Model
{
    use HasFactory;

    // 1. Agrega los nuevos campos al fillable para permitir inserciones
    protected $fillable = [
        'subtema_id',
        'titulo',
        'pregunta',
        'solucion',
        'dificultad',
        'tipo_interaccion', // Nuevo
        'contenido_juego',  // Nuevo
    ];

    // 2. IMPORTANTE: Cast para que Laravel convierta el JSON automáticamente a Array
    protected $casts = [
        'contenido_juego' => 'array', 
    ];

    // Relación inversa (Un ejercicio pertenece a un subtema)
    public function subtema()
    {
        return $this->belongsTo(Subtema::class);
    }
}

File: flutter_application_1/lib/models/ejercicio.dart
import 'dart:convert';

class Ejercicio {
  final int id;
  final int subtemaId; 
  final String titulo;
  final String pregunta;
  final String? dificultad;
  final String? solucion;
  
  // NUEVOS CAMPOS
  final String tipoInteraccion; 
  final Map<String, dynamic>? contenidoJuego;

  Ejercicio({
    required this.id,
    required this.subtemaId,
    required this.titulo,
    required this.pregunta,
    this.dificultad,
    this.solucion,
    required this.tipoInteraccion,
    this.contenidoJuego,
  });

  factory Ejercicio.fromJson(Map<String, dynamic> json) {
    return Ejercicio(
      id: json['id'],
      subtemaId: int.parse(json['subtema_id'].toString()),
      titulo: json['titulo'] ?? 'Ejercicio',
      pregunta: json['pregunta'],
      dificultad: json['dificultad'],
      solucion: json['solucion'],
      // Mapeo de los nuevos campos con valores por defecto
      tipoInteraccion: json['tipo_interaccion'] ?? 'texto_libre',
      contenidoJuego: json['contenido_juego'] is String 
          ? jsonDecode(json['contenido_juego']) 
          : json['contenido_juego'],
    );
  }
}